src/main/java/fi/aalto/cs/apluscourses/intellij/activities/InitializationActivity.java

```diff
@@ -75,6 +80,19 @@ public void runActivity(@NotNull Project project) {
       return;
     }
 
+    var versionComparison =
+        BuildInfo.INSTANCE.courseVersion.compareTo(course.getCourseRequiredVersion());
+
+    if (versionComparison == Version.ComparisonStatus.MAJOR_TOO_OLD
+        || versionComparison == Version.ComparisonStatus.MAJOR_TOO_NEW) {
+      notifier.notify(
+          versionComparison == Version.ComparisonStatus.MAJOR_TOO_OLD
+          ? new CourseVersionOutdatedError() : new CourseVersionTooNewError(), project);
+      return;
+    } else if (versionComparison == Version.ComparisonStatus.MINOR_TOO_OLD) {
+      notifier.notify(new CourseVersionOutdatedWarning(), project);
+    }
```

Comment:
This logic is duplicated. My suggestion: create a factory method that takes a comparison status and returns an appropriate notification.

Then something like:

```java
if (!comparisonStatus.isFullyCompliant())
  notifier.notify(notificationFactory.create(comparisonStatus));

if (!comparisonStatus.isBreaking())
  return;
```